client/src/fabric/tensegrity-fabric.ts

```diff
@@ -247,7 +261,42 @@ export class TensegrityFabric {
 
     public iterate(ticks: number): boolean {
         this.disposeOfGeometry()
-        return this.exports.iterate(ticks)
+        const changeHappened = this.exports.iterate(ticks)
+        if (!changeHappened) {
+            return false
+        }
+        const growth = this.growth
+        if (growth) {
+            if (growth.growing.length > 0) {
+                growth.growing = executeGrowthTrees(growth.growing)
+                this.exports.centralize()
+            }
+            if (growth.growing.length === 0) {
+                if (growth.optimizationStack.length > 0) {
+                    const optimization = growth.optimizationStack.pop()
+                    switch (optimization) {
+                        case "L":
+                            optimizeFabric(this, false)
+                            break
+                        case "H":
+                            optimizeFabric(this, true)
+                            break
+                        case "X":
+                            this.setGestating(3)
+                            growth.optimizationStack.push("Connect")
+                            break
+                        case "Connect":
```

Comment:
Be nice to make these strings constants/enums somewhere and add comments

Reply:
Just haven't taken it serious enough to refine so far.

Reply:
:+1: